### PR TITLE
Simplified tests using demo instance of database that already contains data

### PR DESCRIPTION
**Pull request by northwestwitch**
### This PR adds | fixes:
- The demo database now contains all types of test data. This data can be used directly in tests without the need to load data in tests first. This allows removing many lines of code, that now become redundant.

### How to test:
- Automatic tests

### Expected outcome:
- [x] All tests should pass

### Review:
- [x] Code approved by
- [x] Tests executed by GitHub actions

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -113,6 +97,28 @@ def _override_get_db():
     return TestClient(app)
 
 
+@pytest.fixture(name="demo_session", scope="function")
+def demo_session_fixture() -> TestClient:
+    """Returns an object of type sqlalchemy.orm.session.sessionmaker containing demo data.."""
```
Comment (henrikstranneheim) [inline tests/conftest.py:102]:
```suggestion
    """Returns an object of type sqlalchemy.orm.session.sessionmaker containing demo data."""
```

File: tests/conftest.py [hunk 2]
```diff
@@ -113,6 +97,28 @@ def _override_get_db():
     return TestClient(app)
 
 
+@pytest.fixture(name="demo_session", scope="function")
```
Comment (henrikstranneheim) [inline tests/conftest.py:100]: Could it be scoped to module so that you only have to load once?

**Review by henrikstranneheim — APPROVED** [review]
WOW great clean up💯

File: tests/conftest.py (hunk 2 above)
Comment (northwestwitch) [reply tests/conftest.py:100]: I think this way is safer so a change that you make in a test doesn't affect the test after

Comment (northwestwitch) [reply tests/conftest.py:100]: But if we see that it takes a long time then we can fins another way, of course!

File: tests/conftest.py [hunk 3]
```diff
@@ -113,6 +97,28 @@ def _override_get_db():
     return TestClient(app)
 
 
+@pytest.fixture(name="demo_session", scope="module")
```
Comment (northwestwitch) [inline tests/conftest.py:100]: Using the "module" scope works, tested locally first, so I've changed it as you suggested @henrikstranneheim

**Pull request merged**